web/App_Code/LcRest/JobTitle.cs

```diff
@@ -0,0 +1,495 @@
+                if (statusID == -1)
+                {
+                    return false;
+                }
+
+                // If status is of type 3 'private profile, manual activation'..
+                if (statusID == 3)
+                {
+                    // ..modify the profile StatusID from 3 to 2 'private profile, automatic activation'
+                    // (because the TestProfileActivation only works for StatusID:2 to avoid unexpected changes)
+                    db.Execute(@"
+                            UPDATE  UserProfilePositions
+                            SET     StatusID = 2,
+                                    UpdatedDate = getdate()
+                            WHERE UserID = @0 AND PositionID = @1
+                                AND LanguageID = @2
+                                AND CountryID = @3
+                        ", userID, jobTitleID,
+                        LcData.GetCurrentLanguageID(), LcData.GetCurrentCountryID());
+                }
+
+                // It executes the profile activation that checks that all required alerts must be off to activate:
+                db.Execute("EXEC TestProfileActivation @0, @1", userID, jobTitleID);
+
+                // Check the updated StatusID
+                var newStatusID = LcData.UserInfo.GetUserPositionStatus(userID, jobTitleID);
+                // If the status is 1 'public profile', success!
+                if (newStatusID == 1)
+                {
+                    return true;
+                }
+                else
+                {
+                    // It is Not activated still, required alerts are pending, back to the original
+                    // StatusID if was not 2 originally (it was updated in the middle of the process to run
+                    // the TestProfileActivation procedure)
+                    if (statusID >= 2)
+                    {
+                        db.Execute(@"
+                                UPDATE UserProfilePositions
+                                SET     StatusID = @2,
+                                        UpdatedDate = getdate()
+                                WHERE UserID = @0 AND PositionID = @1
+                                    AND LanguageID = @3
+                                    AND CountryID = @4
+                            ", userID, jobTitleID, statusID,
+                            LcData.GetCurrentLanguageID(), LcData.GetCurrentCountryID());
+                    }
+
+                    return false;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Allows users to create a new job title with basic and default information. #650
+        /// </summary>
+        /// <param name="singularName"></param>
+        /// <param name="languageID"></param>
+        /// <param name="countryID"></param>
+        /// <returns></returns>
+        public static int CreateByName(string singularName, int languageID, int countryID, int enteredByUserID)
```

Comment:
This can go into a JobTitle.cs class because is related to that model not the user-jobtitle. I know, originally all that was mixed at LcData.JobTitle.cs and that was my mistake.
We have a PublicJobTitle.cs class right now, built with the purpose of being a special set of JobTitle fields/features publicly available to users (versus internal details). Can go there too.